providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/targeting/Fractional.java

```diff
@@ -0,0 +1,127 @@
+package dev.openfeature.contrib.providers.flagd.resolver.process.targeting;
+
+import dev.openfeature.contrib.providers.flagd.resolver.process.targeting.lib.Murmur3;
+import io.github.jamsesso.jsonlogic.JsonLogicException;
+import io.github.jamsesso.jsonlogic.evaluator.JsonLogicEvaluationException;
+import io.github.jamsesso.jsonlogic.evaluator.expressions.PreEvaluatedArgumentsExpression;
+import lombok.Getter;
+import lombok.extern.java.Log;
+
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.logging.Level;
+
+@Log
+class Fractional implements PreEvaluatedArgumentsExpression {
+
+    public String key() {
+        return "fractional";
+    }
+
+    public Object evaluate(List arguments, Object data) throws JsonLogicEvaluationException {
+        if (arguments.size() < 2) {
+            return null;
+        }
+
+        final Operator.FlagProperties properties = new Operator.FlagProperties(data);
+
+        // check optional string target in first arg
+        Object arg1 = arguments.get(0);
```

Comment:
I don't think we want to suppor this anymore. This should no longer be a string, and instead be an expression, see here: https://flagd.dev/concepts/#fractional-evaluation

Reply:
Here is the full spec for this: https://github.com/open-feature/flagd/blob/main/docs/other_resources/in-process-providers/evaluators/fractional_evaluation.md

Reply:
In case there is an expression, the value we receive at index 0 (argument 1) is already parsed by the JsonLogic library. This is done at `PreEvaluatedArgumentsExpression` interface.

Reply:
And this is identical to flagd implementation - https://github.com/open-feature/flagd/blob/main/core/pkg/eval/fractional_evaluation.go#L55-L63

Reply:
>  I think by the time this code runs, { "var": "someValue" } will be replaced WITH a string

Exactly this .!! 😃

Reply:
Yes exactly, we receive replaced values. As you observed in flagd [^1], we cannot differentiate whether this value comes through an expression or is replaced by JsonLogic. And the unit tests here only focus on the logical flow proposed in evaluator documentation [^2].

Anyway, good that we discussed this point so it's clear for everyone :) 

[^1]: https://github.com/open-feature/flagd/issues/871 
[^2]: https://github.com/open-feature/flagd/blob/main/docs/other_resources/in-process-providers/evaluators/fractional_evaluation.md

Reply:
Thanks!